cli/tools/coverage/ignore_directives.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use deno_ast::swc::common::comments::Comment;
+use deno_ast::swc::common::comments::CommentKind;
+use deno_ast::view as ast_view;
+use deno_ast::RootNode as _;
+use deno_ast::SourceRange;
+use deno_ast::SourceRanged;
+use deno_ast::SourceRangedForSpanned as _;
+use deno_ast::SourceTextInfoProvider as _;
+use deno_core::url::Url;
+use std::collections::HashMap;
+
+static COVERAGE_IGNORE_START_DIRECTIVE: &str = "deno-coverage-ignore-start";
+static COVERAGE_IGNORE_STOP_DIRECTIVE: &str = "deno-coverage-ignore-stop";
+static COVERAGE_IGNORE_NEXT_DIRECTIVE: &str = "deno-coverage-ignore-next";
+static COVERAGE_IGNORE_FILE_DIRECTIVE: &str = "deno-coverage-ignore-file";
+
+pub type RangeIgnoreDirective = IgnoreDirective<Range>;
+pub type NextIgnoreDirective = IgnoreDirective<Next>;
+pub type FileIgnoreDirective = IgnoreDirective<File>;
+
+pub enum Range {}
+pub enum Next {}
+pub enum File {}
+pub trait DirectiveKind {}
+impl DirectiveKind for Range {}
+impl DirectiveKind for Next {}
+impl DirectiveKind for File {}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct IgnoreDirective<T: DirectiveKind> {
+  range: SourceRange,
+  _marker: std::marker::PhantomData<T>,
+}
+
+impl<T: DirectiveKind> IgnoreDirective<T> {
+  pub fn range(&self) -> SourceRange {
+    self.range
+  }
+}
+
+pub fn parse_range_ignore_directives(
+  is_quiet: bool,
+  script_module_specifier: &Url,
+  program: &ast_view::Program,
+) -> Vec<RangeIgnoreDirective> {
+  let mut depth: usize = 0;
+  let mut directives = Vec::<RangeIgnoreDirective>::new();
+  let mut current_range: Option<SourceRange> = None;
+
+  let mut comments_sorted = program
+    .comment_container()
+    .all_comments()
+    .collect::<Vec<_>>();
+  comments_sorted.sort_by(|a, b| a.range().start.cmp(&b.range().start));
+
+  for comment in comments_sorted.iter() {
+    if comment.kind != CommentKind::Line {
+      continue;
+    }
```

Comment:
Yes, since you can't ignore part of a line in the current implementation. If we removed this condition, the following would perhaps cause unexpected behaviour as the whole line would be ignored, not just the enclosed section:

```ts
function foobar(condition: boolean) {
  return condition ? foo() : /* deno-ignore-coverage-start */ bar() /* deno-ignore-coverage-end */;
}
```